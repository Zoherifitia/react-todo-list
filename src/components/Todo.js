
export const Todo = ({ todo, index, markTodo }) => {
    return (
      <div className="todo">
        <div className='input'>
        <input
          type="checkbox"
          id={index}
          onClick={() => markTodo(index)}
        />
        <label htmlFor={index} />
        <span>{todo.text}</span>
        </div>
      </div>
    );
  }