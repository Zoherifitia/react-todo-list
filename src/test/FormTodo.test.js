import React from 'react';
import renderer from 'react-test-renderer';
import { FormTodo } from "../components/FormTodo"

describe('FormTodo component', () => {
  it('renders correctly', () => {
    const addTodo = jest.fn();
    const tree = renderer
      .create(<FormTodo addTodo={addTodo} />)
      .toJSON();
    console.log(tree)
    expect(tree).toMatchSnapshot();
  });
});