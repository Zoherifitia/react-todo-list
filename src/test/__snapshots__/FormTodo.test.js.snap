// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTodo component renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h3>
    Todo
  </h3>
  <input
    className="input_todo"
    onChange={[Function]}
    placeholder="Add new todo"
    type="text"
    value=""
  />
</form>
`;
